### PR TITLE
Issue 708 - Stacey BC implementation for poroelastic 

### LINKED ISSUE #708: Add boundary conditions for poroelastic
## Task description

Add Stacey, Dirichlet, and composite boundaries for poroelastic domains

- [ ] Implement respective BCs 
- [ ] Add displacement test for every single BC

## Acceptance criteria
- [ ] The new tests pass

### PR DESCRIPTION
**Pull request by Rohit-Kakodkar**
## Description

Implements Stacey BCs for poroelastic domains.

Note: Sound velocity for PE vicious domains is dependent on the frequency of the source. However, we approximate the sound velocity to be sound velocity when `eta = 0`, i.e., there is no viscous damping. 

Note: @cmorency1, we use the same function for computing Stacey contribution to acceleration and mass matrix. Within the mass matrix function, we pass `velocity = -dt/2`

## Issue Number

closes #708 

## Checklist

Please make sure to check developer documentation on specfem docs.

- [ ] I ran the code through pre-commit to check style
- [ ] My code passes all the integration tests
- [ ] I have added sufficient unittests to test my changes
- [ ] I have added/updated documentation for the changes I am proposing
- [ ] I have updated CMakeLists to ensure my code builds
- [ ] My code builds across all platforms

### CODE REVIEW

File: include/boundary_conditions/stacey/stacey.tpp [hunk 1]
```diff
@@ -246,6 +250,154 @@ impl_enforce_traction(const acoustic_type &, const isotropic_type &,
...
+  Kokkos::Experimental::where(mask, traction(2)) =
+      traction(2) +
+      static_cast<type_real>(-1.0) * tfx * jacobian1d * boundary.edge_weight;
+
+  Kokkos::Experimental::where(mask, traction(3)) =
+      traction(3) +
+      static_cast<type_real>(-1.0) * tfz * jacobian1d * boundary.edge_weight;
+
+  return;
+}
```
Comment (lsawade) [inline include/boundary_conditions/stacey/stacey.tpp:399]: Trusting you here...

**Review by lsawade — APPROVED** [review]
LGTM. I guess since we do not have the frequency on the boundary, we can't test against fortran sf2d

**Comment by Rohit-Kakodkar** [comment]
> LGTM. I guess since we do not have the frequency on the boundary, we can't test against fortran sf2d

Yes, and the other reason is - mass matrix computation for poroelastic Stacey is incorrect in 2D Fortran.

**Review by icui — APPROVED** [review]
